RELEASE.md

```diff
@@ -0,0 +1,10 @@
+# Release Checklist
```

Comment:
You could potentially encode this into a [issue template](https://github.com/azavea/vanguard-charitable-charity-search/blob/develop/.github/ISSUE_TEMPLATE/RELEASE.md), if that makes the workflow easier for others to follow.

Reply:
Good suggestion. I thought about it a bit, and for now I think it's simpler just to keep the checklist. It looks like if you add issue template just for release you get something that looks like this:

![Screen Shot 2020-11-04 at 9 28 34 AM](https://user-images.githubusercontent.com/1818302/98123640-1f04fd80-1e80-11eb-8c0d-4e84fcd6b01a.png)

Which means, because of GitHub's UI, I wouldn't feel comfortable providing _just_ a release template, because they slam that tiny little "open a blank issue" link in the bottom right. 😡 

It's not too much extra work to add a bug and feature report template, but this is such a small repo that it feels heavy handed.